### DISCUSSION TITLE
SiriusXM as a music provider

### DISCUSSION THREAD

**Discussion by dacmcbibs** [music-assistant/support#1161] [Feature requests and ideas]
Would it be possible to add SiriusXM as a music provider ?
Maybe these GitHub projects can help:
https://github.com/andrew0/SiriusXM
https://github.com/schlarpc/SeriousCast

**Comment by greggotcher** [comment]
I would love to see this as a provider

**Comment by sspaul1976** [comment]
+1 vote from me as well

**Comment by TheWebMachine** [comment]
+1 from me!

**Comment by madbrain76** [comment]
I would like to see this as well.

**Comment by lpl-maa** [comment]
+1

**Comment by rbonser** [comment]
+1

**Comment by PapiOrtiz** [comment]
+1

**Comment by CG821** [comment]
+1

**Comment by balones6531** [comment]
+1

**Comment by OzGav** [comment]
Siriusxm is now available in beta 32

**Reply by marcelveldt** [reply]
correction: beta 33

**Reply by ColtonYYZ** [reply]
how do we download it?  When I go into Hacs and click on Download, I don't see beta 32 or 33. I see 2024.7.1 beta 0 from 3 months ago, then it goes 8.0, 8.1, 9.0, 9.1, 10.0, 10.1

**Reply by OzGav** [reply]
It’s in the beta version of the server not the integration

**Reply by OzGav** [reply]
Update your integration while you are there though

**Reply by ColtonYYZ** [reply]
> It’s in the beta version of the server not the integration

Okay, I was doing it in Hacs (DUH! I needed to add it in Add-ons)... so I configured SiriusXM and added my Home Assistant media players. When I select a channel and add it to the queue (I Selected Play Now), and hit play, I get the male Alexa voice saying, "To send TTS, please set announce to 'True'. Music cannot be played this way."

Did I miss something when configuring my SiriusXM account and media players?

**Reply by OzGav** [reply]
You cannot stream to Alexa

**Reply by ColtonYYZ** [reply]
I'm just clueing in now as I commented on that discussion about MyMedia being able to play mp3's longer than 4 minutes and the fact that I can tell Alexa to play SiriusXM on my Echo's.

How can do I some testing for you between the two? Would it even be helpful? I'm a noob at this, but love a challenge and rarely give up. hehe

**Reply by OzGav** [reply]
Sorry I am not following what you are asking. Playing a streaming URL to Alexa does not currently work. Discussion about Alexa is here https://github.com/orgs/music-assistant/discussions/431

**Reply by ColtonYYZ** [reply]
Yeah, my comment is [here](https://github.com/orgs/music-assistant/discussions/431#discussioncomment-10851057).

I was saying that I can play SiriusXM on my Alexa devices with the SiriusXM app, and use voice commands with Alexa to play whatever channel I want to from Sirius. So Alexa devices must have a way to play them.  

And MyMedia Alexa can play mp3's that are longer than 4 mins which is the limit when using Home Assistant to send the notify command, or using Chime_TTS to play mp3's over Alexa devices. 

I contacted the creator of MyMedia Alexa to see if he could provide any insight since he got his program to work with Alexa devices but he said he was rather busy at that time, but would take a look eventually.

I have Alexa devices and I also pay for MyMedia Alexa, so if there's some way of testing of sort that I can do to help, I'm more than happy to. I'd just need some guidance on how to do the testing if it's needed, is all.

**Reply by OzGav** [reply]
This is not a question about SiriusXM it is about playing to Alexa which as I said does not currently work and you can read all about why in that thread I linked

**Reply by ColtonYYZ** [reply]
I'm well aware what it's about. I know how to read. There seems to be a misunderstanding as to what I was offering. It's fine. Thanks for your time.

**Comment by sspaul1976** [comment]
Awesome! Thanks for adding this.

**Comment by dacmcbibs** [comment]
Agreed, super excited to try this out tonight!

**Reply by dacmcbibs** [reply]
Its working great!

**Comment by TheWebMachine** [comment]
Haha, of course, right after I finally get around to canceling SXM to avoid yet another annual charge. Enjoy!

**Comment by murfnturf1** [comment]
Tried to add SXM provider. Might have mistyped my password and it did not authenticate. It does not show up in my list of providers. Tried adding the account again but it gives me a warning that the provider is already loaded and only one instance is allowed. There is a blank SiriusXM folder when click browse in the lower right.

**Reply by OzGav** [reply]
https://github.com/music-assistant/hass-music-assistant/issues/3127

**Reply by murfnturf1** [reply]
Thanks, sorry I did not see that post when I searched. Will follow that thread for updates.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
